includes/options/class-amp-options-menu.php

```diff
@@ -423,6 +436,191 @@ private function list_template_conditional_options( $options, $parent = null ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		$plugins = get_plugins();
+		unset( $plugins['amp/amp.php'] );
+		foreach ( array_keys( $plugins ) as $plugin_file ) {
+			if ( ! is_plugin_active( $plugin_file ) ) {
+				unset( $plugins[ $plugin_file ] );
+			}
+		}
+		uasort(
+			$plugins,
+			static function ( $a, $b ) {
+				return strcmp( $a['Name'], $b['Name'] );
+			}
+		);
+
+		?>
+		<fieldset>
+			<h4 class="title hidden"><?php esc_html_e( 'Suppressed Plugins', 'amp' ); ?></h4>
+			<p>
+				<?php esc_html_e( 'When a plugin emits invalid markup that causes an AMP validation error, one option is to review the invalid markup and allow it to be removed. Another option is to suppress the plugin from doing anything when rendering AMP pages. What follows is the list of active plugins with any causing validation errors being highlighted. If a plugin is emitting invalid markup that is causing validation errors and this plugin is not necessary on the AMP version of the page, it can be suppressed.', 'amp' ); ?>
+			</p>
+
+			<style>
+			.amp-suppressed-plugins .suppressed-plugin:checked + label {
+				text-decoration: line-through;
+			}
+			.amp-suppressed-plugins .plugin > details {
+				margin-left: 30px;
+			}
+			.amp-suppressed-plugins .plugin > details > ul {
+				margin-left: 30px;
+				margin-top: 0.5em;
+				margin-bottom: 1em;
+				list-style-type: disc;
+			}
+			.amp-suppressed-plugins summary {
+				user-select: none;
+				cursor: pointer;
+			}
+			</style>
+
+			<?php
+			$element_name = AMP_Options_Manager::OPTION_NAME . '[' . Option::SUPPRESSED_PLUGINS . ']';
+
+			$errors_by_sources = self::get_plugin_errors_by_sources()
+			?>
+			<ul>
+				<?php foreach ( $plugins as $plugin_file => $plugin ) : ?>
+					<?php
+					$plugin_slug = strtok( $plugin_file, '/' );
+
+					$is_suppressed = array_key_exists( $plugin_slug, $suppressed_plugins );
+					if ( ! $is_suppressed && ! isset( $errors_by_sources[ $plugin_slug ] ) ) {
+						continue;
+					}
+					?>
+					<li class="plugin">
+						<input
+							type="checkbox"
+							class="suppressed-plugin"
+							id="<?php echo esc_attr( "$element_name-$plugin_file" ); ?>"
+							name="<?php echo esc_attr( $element_name . '[]' ); ?>"
+							value="<?php echo esc_attr( $plugin_slug ); ?>"
+							<?php checked( $is_suppressed ); ?>
+						>
+						<label for="<?php echo esc_attr( "$element_name-$plugin_file" ); ?>">
+							<?php
+							if ( ! $is_suppressed ) {
+								echo '<strong>';
+							}
+							echo esc_html( $plugin['Name'] );
+							if ( ! $is_suppressed ) {
+								echo '</strong>';
+							}
+							?>
+						</label>
+						<?php if ( $is_suppressed && version_compare( $suppressed_plugins[ $plugin_slug ][ Option::SUPPRESSED_PLUGINS_LAST_VERSION ], $plugins[ $plugin_file ]['Version'], '!=' ) ) : ?>
+							<small>
+								<?php if ( $suppressed_plugins[ $plugin_slug ][ Option::SUPPRESSED_PLUGINS_LAST_VERSION ] && $plugins[ $plugin_file ]['Version'] ) : ?>
```

Comment:
Should this condition be checked before comparing versions above?

Reply:
This is here in the rare situation where a plugin was updated to add a version which previously lacked it, or it was updated to remove the version where it previously had one.